### DISCUSSION TITLE
Query regarding Node2Vec model

### DISCUSSION THREAD

**Discussion by shrimonmuke0202** [pyg-team/pytorch_geometric#6841] [Q&A]
Hi Team,

I have some molecular graphs. I want to generate node features using the node2vec model for that molecular graph, where the node2vec model weights are trained during the training. How can I do this? I will be very grateful If you answered it will be very helpful for me.

**Comment by EdisonLeeeee** [comment]
Does that [example](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/node2vec.py) work for you?

**Comment by shrimonmuke0202** [comment]
The [example](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/node2vec.py) contains a single graph, but I have multiple graphs which are molecules. So, this [example](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/node2vec.py)  does not works for me.

**Reply by EdisonLeeeee** [reply]
You can perform node2vec for every single molecular graph separately.

**Reply by rusty1s** [reply]
Yeah, that's the only workaround. There is currently no way to use `Node2Vec` on a set of graphs since it wasn't really designed for this use case.

Please note that I don't think it is feasible to use node2vec features across multiple graphs since the embeddings will be totally different each time.
